### PR TITLE
How to publish an app to the app stores

### PR DESCRIPTION
**Pull request by MonicaG**
This page is a rewrite of the [Apple API Key Usage](https://github.com/bcgov/mobile-signing-service/wiki/Apple-API-Key-Usage) page. That page was focused on how a member of the Developer Experience team would create an API Key. I rewrote the page to focus on the user.

### CODE REVIEW

File: docs/publish.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Publish your app
```
Comment (kaitlyn-rosenburg) [inline docs/publish.md:1]: add in intro sentence describing this page

File: docs/publish.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Distribution methods
+
+See the [Mobile App Distribution Methods](distribution_methods.md)  page for an overview of app distribution.
```
Comment (kaitlyn-rosenburg) [inline docs/publish.md:5]:
```suggestion
Review [Mobile App Distribution Methods](distribution_methods.md) for an overview of app distribution.
```

Comment (kaitlyn-rosenburg) [reply docs/publish.md:5]: we can simply link to other pages, we don't need to say "see this other page"

File: docs/publish.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## CI/CD
+
+Let the [Developer Experience](contact.md) team know your app will use a CI/CD pipeline to upload your app. We will give you access to the necessary API keys.
```
Comment (kaitlyn-rosenburg) [inline docs/publish.md:9]:
```suggestion
Let the [Developer Experience](contact.md) team know if your app will use a CI/CD pipeline to upload your app. We'll give you access to the necessary API keys.
```

File: docs/publish.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+See the [bc-wallet-mobile](https://github.com/bcgov/bc-wallet-mobile/) project as an example of an app using GitHub Actions to upload to both app stores.
```
Comment (kaitlyn-rosenburg) [inline docs/publish.md:11]:
```suggestion
 The [bc-wallet-mobile](https://github.com/bcgov/bc-wallet-mobile/) project is a good example of an app using GitHub Actions to upload to both app stores.
```

File: docs/publish.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+## App Store Connect
+
+Let us know if you plan to:
+
+1. upload the build, or,
```
Comment (kaitlyn-rosenburg) [inline docs/publish.md:17]:
```suggestion
1. Upload the build
```

File: docs/publish.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+1. upload the build and ship it to TestFlight
```
Comment (kaitlyn-rosenburg) [inline docs/publish.md:18]:
```suggestion
or
1. Upload the build and ship it to TestFlight
```

File: docs/publish.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+The answer to the above will determine which API Key to use.
```
Comment (kaitlyn-rosenburg) [inline docs/publish.md:20]:
```suggestion
The upload path you pick determines which API Key to use. 
```

File: docs/publish.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+The bcgov GitHub organization stores the [App Store Connect API](https://developer.apple.com/documentation/appstoreconnectapi) keys as secrets. Send the [Developer Experience](contact.md) team your repo name and we will give it access to those secrets.
```
Comment (kaitlyn-rosenburg) [inline docs/publish.md:22]:
```suggestion
The bcgov GitHub organization stores the [App Store Connect API](https://developer.apple.com/documentation/appstoreconnectapi) keys as secrets. Send the [Developer Experience](contact.md) team your repo name and we'll give it access to those secrets.
```

File: docs/publish.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+You can use [codemagic-cli-tools](https://docs.codemagic.io/knowledge-codemagic/codemagic-cli-tools/) to upload your app. See the [bc-wallet-mobile](https://github.com/bcgov/bc-wallet-mobile/) project as an example.
```
Comment (kaitlyn-rosenburg) [inline docs/publish.md:24]:
```suggestion
You can use [codemagic-cli-tools](https://docs.codemagic.io/knowledge-codemagic/codemagic-cli-tools/) to upload your app. Use the [bc-wallet-mobile](https://github.com/bcgov/bc-wallet-mobile/) project as an example.
```

Comment (kaitlyn-rosenburg) [reply docs/publish.md:24]: might be cleaner to just mention the BC Wallet example once

File: docs/publish.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+## Play Console
+See the [Developer Experience](contact.md) team to setup your API Key to the Play Console.
```
Comment (kaitlyn-rosenburg) [inline docs/publish.md:27]:
```suggestion
Contact the [Developer Experience](contact.md) team to setup your API Key to the Play Console.
```

File: docs/publish.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+You can use the `npx @bcgov/gpublish` package to upload your app. See the [bc-wallet-mobile](https://github.com/bcgov/bc-wallet-mobile/) project as an example.
```
Comment (kaitlyn-rosenburg) [inline docs/publish.md:29]:
```suggestion
You can use the `npx @bcgov/gpublish` package to upload your app. 
```

Comment (kaitlyn-rosenburg) [reply docs/publish.md:29]: see above comment. can just mention the example once

**Review by kaitlyn-rosenburg — CHANGES_REQUESTED** [review]

File: docs/publish.md (hunk 1 above)
Comment (MonicaG) [reply docs/publish.md:1]:
```suggestion
# Publish your app

This page describes how to setup your CI/CD pipeline to upload your app to App Store Connect and Play Console.
```

File: docs/publish.md (hunk 9 above)
Comment (MonicaG) [reply docs/publish.md:24]:
```suggestion
You can use [codemagic-cli-tools](https://docs.codemagic.io/knowledge-codemagic/codemagic-cli-tools/) to upload your app. 
```

Comment (MonicaG) [reply docs/publish.md:24]: removed mention of BC Wallet as suggested

**Comment by MonicaG** [comment]
@kaitlyn-rosenburg - I've updated the [page with your suggestions](https://github.com/bcgov/mobile-developer-guide/pull/17/commits/e306316c9039df486982fcdaedbe0131235f74ff). I also realized I didn't update the navigation file to link to this page. I've made that update and you can [review that update here](https://github.com/bcgov/mobile-developer-guide/pull/17/commits/83168bdd3108624e41e994e321f899fa91fb5805).

File: docs/publish.md [hunk 12]
```diff
@@ -1,29 +1,32 @@
 # Publish your app
 
+This page describes how to setup your CI/CD pipeline to upload your app to App Store Connect and Play Console.
```
Comment (kaitlyn-rosenburg) [inline docs/publish.md:3]: change to: Learn how to setup your CI/CD pipeline to upload your app to App Store Connect and Play Console.

**Review by kaitlyn-rosenburg — CHANGES_REQUESTED** [review]

**Pull request merged**